src/presentation/NewCompositionRoot.ts

```diff
@@ -20,12 +25,18 @@ function getCompositionRoot(repositories: Repositories) {
             getStorageClient: new GetStorageClientUseCase(repositories.storageClientRepository),
             setStorageClient: new SetStorageClientUseCase(repositories.storageClientRepository),
         },
+
+        settings: {
```

Comment:
Be careful here.

Pass all repositories as object is less readable using structural typing because here I don't know that repositories are passed to the use cases. I need to enter in the use case to review.

I like the code to be clear. Here using structural typing in runtime pass all repositories 

 if you want to maintain repositories as object in the use cases, it is more readable and safe to create a new object and pass it